pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -993,3 +976,32 @@ func (m *kubeGenericRuntimeManager) removeContainerLog(containerID string) error
 func (m *kubeGenericRuntimeManager) DeleteContainer(containerID kubecontainer.ContainerID) error {
 	return m.removeContainer(containerID.ID)
 }
+
+// setTerminationGracePeriod determines the grace period to use when killing a container
+func setTerminationGracePeriod(pod *v1.Pod, containerSpec *v1.Container, reason containerKillReason) int64 {
+	if pod.DeletionGracePeriodSeconds != nil {
```

Comment:
I believe the switch statement from the old code looks and functions cleaner than the multiple if statements. Can you refactor this function again to use the switch?

Reply:
done... also added a helper function to reduce some of the code duplication